services/rsources/stats_collector.go

```diff
@@ -286,3 +269,47 @@ func (r *statsCollector) buildStats(jobs []*jobsdb.JobT, failedJobs map[uuid.UUI
 		}
 	}
 }
+
+type parametersParser func(jp json.RawMessage) (jobRunID, recordID string, target JobTargetKey)
+
+type OptFunc func(*statsCollector)
+
+// IgnoreDestinationID ignores the destinationID parameter of the job and while capturing statistics
+func IgnoreDestinationID() OptFunc {
+	return func(r *statsCollector) {
+		r.parametersParser = func(jobParams json.RawMessage) (jobRunID, recordID string, target JobTargetKey) {
+			jobRunID, recordID, target = defaultJobIdentifier(jobParams)
+			target.DestinationID = ""
+			return jobRunID, recordID, target
+		}
+	}
+}
+
+func defaultJobIdentifier(jobParams json.RawMessage) (jobRunID, recordID string, target JobTargetKey) {
+	var jobRunId string
+	var jobTargetKey JobTargetKey
+	var recordId string
+	remaining := 5
+	jp := gjson.ParseBytes(jobParams)
```

Comment:
Any benefit of doing ParseBytes instead of unmarshaling?

Reply:
it is iterating over the field tokens, so we can gain some iterations as soon as we are done with the fields we are interested in (performance benefit)